packages/core/src/sdk/analytics/index.tsx

```diff
@@ -12,14 +12,20 @@ if (typeof window !== 'undefined') {
 }
 
 const AnalyticsHandler = (): null => {
-  useAnalyticsEvent((event: AnalyticsEvent) => {
+  useAnalyticsEvent((event: UnknownEvent) => {
     // Cleans the ecommerce object before pushing a new one
     // This prevents the new data from getting merged with the previous one
     // which could lead do inaccurate and old data being sent with events
-    //
+
     // source: https://developers.google.com/tag-manager/ecommerce-ga4?hl=pt-br#clearing_the_ecommerce_object
+
     window.dataLayer.push({ ecommerce: null })
-    window.dataLayer.push({ event: event.name, ecommerce: event.params })
+
+    const payload = event.isEcommerceEvent
+      ? { event: event.name, ecommerce: event.params }
+      : { event: event.name, ...event.params }
```

Comment:
nitpicking: just a type assertion to keep the type as before.
```suggestion
    const payload = event.isEcommerceEvent
      ? ({ event: event.name, ecommerce: event.params } as {
          event: AnalyticsEvent['name']
          ecommerce: AnalyticsEvent['params']
        })
      : { event: event.name, ...event.params }
```